### ISSUE TITLE
Feature request: Allow sorting by other criteria

### ISSUE THREAD

**Issue by tungstonminer** [cpw/inventorysorter#122]
In particular, I'd like to be able to sort an inventory by display name rather than stack size.  However, one could easily imagine sorting by a variety of things (e.g., similar to how NotEnoughItems  allows many sorting by various criteria.

**Comment by ShamblerDK** [comment]
Yes, please. I came to this issue tracker for that exact reason. Sorting by anything other than name is just extremely unintuitive.

**Comment by cpw** [comment]
Size -> Name is the current sorting mechanism...

**Comment by Zerahn** [comment]
The request is to change it to (or allow the option of) name -> stack size. I was tossing stuff into a chest and resorting and I couldn't figure out why things were changing their positions until it dawned on me that stack size is the primary criteria. It makes it hard to find specific things when some of them look the same, especially when the order might as well be random.

**Comment by DZK312** [comment]
> Yes, please. I came to this issue tracker for that exact reason. Sorting by anything other than name is just extremely unintuitive.

Same. It's very unhelpful to sort by stack size/amount exclusively. sorting by ids/types/names/etc would be so much better. like if i sorted a chest of vanilla ores for example, the gold blocks, ingots, and nuggets should be side by side, not blocks at the bottom, and ingots and nuggets wherever the wind blows and the like. I'm kind of surprised it wasn't already a feature.

**Comment by ShadowDrakken** [comment]
What criteria is it even sorting by now? Because I can sort an inventory repeatedly and every time it comes back in a different order. With NO changes to the inventory... I find this highly suspect, and completely useless.

**Comment by BratrilliantG7** [comment]
More sorting options would make this mod _perfect_. The constant changing of the order of my chest anytime I add or remove a few items is hard to keep track of at times, whereas with a name- or id-based sorting system adding a stack of something would simply put it next to the existing stacks instead of moving all of those stack further up in the chest.

That's why I typically prefer Quark's inventory sorting. While I'm not sure what exactly it sorts by (some sort of category system?), I know it's not by quantity, so everything in my chest will remain in the same order regardless of how much is in there. I just dislike adding the whole Quark mod just to disable most of it to keep the sorting.
